### PR TITLE
Udemy scraper

### LINKED ISSUE #862: Udemy courses details scraper
# Aim

To scrape details of Udemy courses.

# Details

Important details about courses like Course title, description, instructor, price, rating and number of lectures will be scrapped and stored in a db

### PR DESCRIPTION
**Pull request by Ayushjain2205**
# Description

Script added to scrape udemy course information and store it in a database

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Udemy Scraper/fetcher.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+# Function to insert into table
+def sql_insert(con, entities):
+    cursorObj = con.cursor()    
+    cursorObj.execute('INSERT INTO courses(title, description, instructor, current_price, original_price, rating, hours, lectures) VALUES(?, ?, ?, ?, ?, ?, ?, ?)', entities)    
+    con.commit()
+ 
+
+# Get chrome driver path
+driver_path = input("Enter chrome driver path")
```
Comment (kaustubhgupta) [inline Udemy Scraper/fetcher.py:35]:
```suggestion
driver_path = input("Enter chrome driver path: ")
```

File: Udemy Scraper/fetcher.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+# Get input for course category to scrape
+category = input("Enter course category:")
```
Comment (kaustubhgupta) [inline Udemy Scraper/fetcher.py:38]: Can you display some of the categories that are available on Udemy? A user like me is not aware of different options in category. Display them as a list to the user. Also this minor change:

```suggestion
category = input("Enter course category: ")
```

File: Udemy Scraper/fetcher.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+import time
+import sqlite3
+from sqlite3 import Error
+
+# Function to connect to the SQL Database
+def sql_connection():
+    try:
+        con = sqlite3.connect('udemyDatabase.db')
```
Comment (kaustubhgupta) [inline Udemy Scraper/fetcher.py:12]:
```suggestion
        con = sqlite3.connect('./Udemy Scraper/udemyDatabase.db')
```

File: Udemy Scraper/display.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import sqlite3
+from sqlite3 import Error
+
+# Function to connect to the SQL Database
+def sql_connection():
+    try:
+        con = sqlite3.connect('udemyDatabase.db')
```
Comment (kaustubhgupta) [inline Udemy Scraper/display.py:7]:
```suggestion
        con = sqlite3.connect('./Udemy Scraper/udemyDatabase.db')
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by Ayushjain2205** [comment]
@kaustubhgupta  I have made the suggested changes

**Review by kaustubhgupta — APPROVED** [review]
💯 Good improvement from first PR

**Review by santushtisharma10 — APPROVED** [review]
@Ayushjain2205 💯

**Review by Kushal997-das — APPROVED** [review]
LGTM 👍

**Comment by santushtisharma10** [comment]
@antrikshmisri

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚩

**Pull request merged**